doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,310 @@
+
+  * RLB - rule locking bypass.
+
+    * Modify locked region with RLB set.
+    * Try to enable RLB when RLB is disabled and locked regions present.
+
+  * MMWP - machine mode whitelist policy.
+
+    * M-mode access fail due to not matching any PMP regions.
+    * Try to disable when enabled.
+
+  * MML - machine mode lockdown policy.
+
+    * Try to disable when enabled.
+
+  * Access close to PMP region modification that allows/disallows that access.
+
+CSRs
+^^^^
+Basic read/write functionality must be tested on all implemented CSRs.
+
+* Read from CSR.
+* Write to CSR.
+
+  * Write to read only CSR.
+
+* Write illegal/unsupported value to WARL field.
+* Access to CSR disallowed due to privilege level/debug mode.
+* Read and write from/to an unimplemented CSR
+
+CSRs addresses do not need to be crossed with the variety of CSR instructions as these all use the same basic read & write interface into ``ibex_cs_registers``.
+Coverage of the above points will be sampled at the ``ibex_cs_registers`` interface (as opposed to sampling CSR instructions).
+
+Miscellaneous
+^^^^^^^^^^^^^
+Various points of interest do not fit into the categories above.
+
+* Instruction unstalled - Cover the cycle an instruction is unstalled having just been stalled.
+* Stalled awaiting an instruction fetch.
+* Double fault.
+* Awake from sleep.
+* Interrupt/Debug whilst entering sleep.
+
+Cross Coverage
+--------------
+Much of the more complex behaviour lies at the combination of the individual microarchitectural behaviours above.
+Cross coverage is used to capture that.
+There are some broad crosses containing many bins aiming to capture all combinations of some generalised behaviours as well as some more specific ones to capture all combinations of behaviours focused on a particular area.
+
+Cross coverage will be intentionally broad.
+Where it is proving hard to hit particular bins they will be reviewed in more detail to determine if they're impossible to hit or if simply hard to hit and whether hitting them provides meaningful gains to verification quality.
+
+Excluded bins will either become illegal bins (where they are impossible to hit, so a failure will be seen if they are hit) or ignore bins (where they don't factor into coverage statistics).
+There must be a documented reason a particular bin is added to the illegal or ignore bins.
+
+* All combinations of privilege levels across IF, ID/EX and LSU
+* Pipeline stage states across IF, ID/EX and WB x Instruction Categories
+
+  * All bins containing instruction categories other than **None** ignored when ID/EX stage is empty.
+
+* Instructions Categories x ID/EX Privilege level
+* Instruction Categories x Stall Categories
+
+  * Illegal bins will be used to exclude instruction and stall categories that cannot occur.
+
+* Instruction Categories x Hazards
+* Instruction Categories x Debug Mode
+* Instruction Categories x IF/WB full or empty
```

Comment:
This cross is just the pipeline stage state x instruction categories cross above but with less stuff, I'll remove it.